Library/Homebrew/cmd/update-report.rb

```diff
@@ -167,16 +170,113 @@ def migrate_legacy_cache_if_necessary
     end
   end
 
-  def link_completions_and_docs
-    return if HOMEBREW_PREFIX.to_s == HOMEBREW_REPOSITORY.to_s
+  def migrate_legacy_repository_if_necessary
+    return unless HOMEBREW_PREFIX.to_s == "/usr/local"
+    return unless HOMEBREW_REPOSITORY.to_s == "/usr/local"
+    return unless ARGV.homebrew_developer?
+
+    unless HOMEBREW_PREFIX.writable_real?
+      onoe <<-EOS.undent
+        #{HOMEBREW_PREFIX} is not writable.
+
+        You should change the ownership and permissions of #{HOMEBREW_PREFIX}
+        temporarily back to your user account so we can complete the Homebrew
+        repository migration:
+          sudo chown -R $(whoami) #{HOMEBREW_PREFIX}
+      EOS
+    end
```

Comment:
Let's explain the migration at the front of these error messages. Something like:...?

> Homebrew is rearranging some furniture! Nothing will change about how you use Homebrew, but some files are moving to /usr/local/Homebrew to make the Sierra upgrade smoother. In order for this migration to succeed, ...

(do we think this will make the Sierra upgrade smoother?)